dashboard/app/controllers/api/v1/pd/workshops_controller.rb

```diff
@@ -278,6 +296,23 @@ def potential_organizers
     Pd::WorkshopMailer.organizer_detail_change_notification(@workshop).deliver_now
   end
 
+  private def adjust_course_offerings(supplied_course_offering_ids)
+    existing_course_offering_ids = @workshop.course_offerings.map(&:id)
+    new_course_offering_ids = supplied_course_offering_ids - existing_course_offering_ids
+    course_offering_ids_to_remove = existing_course_offering_ids - supplied_course_offering_ids
+
+    course_offering_ids_to_remove.each do |course_offering_id|
+      @workshop.course_offerings.delete(course_offering_id)
+    end
+
+    new_course_offering_ids.each do |course_offering_id|
+      course_offering = CourseOffering.find_by(id: course_offering_id)
+      # Return if we don't find a corresponding/valid Course Offering
+      next unless course_offering
+      @workshop.course_offerings << course_offering
+    end
```

Comment:
not 100% sure this would work, but you might be able simplify this to 
```
@workshop.course_offerings = supplied_course_offering_ids.map {|id| CourseOffering.find_by(id: course_offering_id) }.compact
```

Reply:
I got a version of this working with filter_map! TIL Rails magic will go through and delete join table associations without me having to do it explicitly:) Thanks!